pkg/controller/datadogagentdeployment/service.go

```diff
@@ -40,7 +40,21 @@ func (r *ReconcileDatadogAgentDeployment) manageClusterAgentService(logger logr.
 		return reconcile.Result{}, err
 	}
 
-	return r.updateIfNeededClusterAgentService(logger, dad, service, newStatus)
+	if result, err := r.updateIfNeededClusterAgentService(logger, dad, service, newStatus); err != nil {
+		return result, err
+	}
+
+	// Make sure the corresponding Endpoints object is created
+	endpoints := &corev1.Endpoints{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: dad.Namespace, Name: serviceName}, endpoints)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{Requeue: true}, fmt.Errorf("cluster agent service endpoints object %s/%s is not created yet", dad.Namespace, serviceName)
+		}
+		return reconcile.Result{}, err
+	}
```

Comment:
I think this is not necessary, since this should be handle automatically by the kubernetes controllers. Also you already test it in the `reconcileClusterAgent` function

Reply:
correct 👍